wannierberri/__system_tb.py

```diff
@@ -72,23 +72,22 @@ def __init__(self,tb_file="wannier90_tb.dat",**parameters):
         self.iRvec=np.array(self.iRvec,dtype=int)
 
         if self.getAA:
-          self.AA_R=np.zeros( (self.num_wann,self.num_wann,nRvec,3) ,dtype=complex)
-          for ir in range(nRvec):
-            f.readline()
-            assert (np.array(f.readline().split(),dtype=int)==self.iRvec[ir]).all()
-            aa=np.array( [[f.readline().split()[2:8] 
+            self.AA_R=np.zeros( (self.num_wann,self.num_wann,nRvec,3) ,dtype=complex)
+            for ir in range(nRvec):
+                f.readline()
+                assert (np.array(f.readline().split(),dtype=int)==self.iRvec[ir]).all()
+                aa=np.array( [[f.readline().split()[2:8] 
                              for n in range(self.num_wann)] 
                                 for m in range(self.num_wann)],dtype=float)
-            self.AA_R[:,:,ir,:]=(aa[:,:,0::2]+1j*aa[:,:,1::2]).transpose( (1,0,2) ) /self.Ndegen[ir]
+                self.AA_R[:,:,ir,:]=(aa[:,:,0::2]+1j*aa[:,:,1::2]).transpose( (1,0,2) ) /self.Ndegen[ir]
+            self.wannier_centers_cart_auto =  np.array( np.diagonal(self.AA_R[:,:,self.iR0,:],axis1=0,axis2=1).transpose() )
```

Comment:
.T instead of transpose. Also do you need np.array?

Reply:
right